docker-entrypoint.sh

```diff
@@ -0,0 +1,447 @@
+            [ -z "${DB_USER}" ] && {
+                log "ERROR: DB_USER is required for ${DB_TYPE} database"
+                exit 1
+            }
+            [ -z "${DB_NAME}" ] && {
+                log "ERROR: DB_NAME is required for ${DB_TYPE} database"
+                exit 1
+            }
+            ;;
+        *)
+            log "ERROR: Unsupported database type: ${DB_TYPE}. Supported types: sqlite, mysql, pgsql"
+            exit 1
+            ;;
+    esac
+    debug_log "Database validation function completed"
+}
+
+# Validate DNS configuration
+validate_dns_config() {
+    debug_log "DNS_NS1='${DNS_NS1}'"
+    debug_log "DNS_NS2='${DNS_NS2}'"
+    debug_log "DNS_HOSTMASTER='${DNS_HOSTMASTER}'"
+    [ -z "${DNS_NS1}" ] && {
+        log "ERROR: DNS_NS1 (primary nameserver) is required"
+        exit 1
+    }
+    [ -z "${DNS_NS2}" ] && {
+        log "ERROR: DNS_NS2 (secondary nameserver) is required"
+        exit 1
+    }
+    [ -z "${DNS_HOSTMASTER}" ] && {
+        log "ERROR: DNS_HOSTMASTER is required"
+        exit 1
+    }
+    debug_log "DNS validation completed"
+}
+
+# Validate mail configuration if enabled
+validate_mail_config() {
+    local mail_enabled=$(echo "${PA_MAIL_ENABLED:-true}" | tr '[:upper:]' '[:lower:]')
+    if [ "$mail_enabled" = "true" ] && [ "${PA_MAIL_TRANSPORT}" = "smtp" ]; then
+        [ -z "${PA_SMTP_HOST}" ] && {
+            log "ERROR: PA_SMTP_HOST is required when using SMTP transport"
+            exit 1
+        }
+        [ -z "${PA_MAIL_FROM}" ] && {
+            log "ERROR: PA_MAIL_FROM is required when mail is enabled"
+            exit 1
+        }
+    fi
+}
+
+# Validate API configuration if enabled
+validate_api_config() {
+    local api_enabled=$(echo "${PA_API_ENABLED:-false}" | tr '[:upper:]' '[:lower:]')
+    if [ "$api_enabled" = "true" ] && [ -n "${PA_PDNS_API_URL}" ]; then
+        [ -z "${PA_PDNS_API_KEY}" ] && {
+            log "ERROR: PA_PDNS_API_KEY is required when PowerDNS API URL is specified"
+            exit 1
+        }
+    fi
+}
+
+# Validate LDAP configuration if enabled
+validate_ldap_config() {
+    local ldap_enabled=$(echo "${PA_LDAP_ENABLED:-false}" | tr '[:upper:]' '[:lower:]')
+    if [ "$ldap_enabled" = "true" ]; then
+        local required_ldap_vars=("PA_LDAP_URI" "PA_LDAP_BASE_DN")
+        for var in "${required_ldap_vars[@]}"; do
+            [ -z "${!var}" ] && {
+                log "ERROR: ${var} is required when LDAP is enabled"
+                exit 1
+            }
+        done
+    fi
+}
+
+# Create initial admin user if specified
+create_admin_user() {
+    local create_admin=$(echo "${PA_CREATE_ADMIN:-false}" | tr '[:upper:]' '[:lower:]')
+
+    if [ "$create_admin" != "true" ] && [ "$create_admin" != "1" ] && [ "$create_admin" != "yes" ]; then
+        debug_log "Admin user creation disabled"
+        return 0
+    fi
+
+    local admin_username="${PA_ADMIN_USERNAME:-admin}"
+    local admin_password="${PA_ADMIN_PASSWORD:-admin}"
```

Comment:
Maybe this password should still be generated if, for example, nothing was provided?